sdcard/rootfs/root/pwnagotchi/scripts/pwnagotchi/ui/display.py

```diff
@@ -196,6 +206,14 @@ def _waveshare_render(self):
         elif self._is_waveshare2():
             self._display.displayPartial(buf)
 
+    def _waveshare_bc_render(self):
+        buf_black = self._display.getbuffer(self.canvas)
+        emptyImage = Image.new('1', (self._display.height, self._display.width), 255)
```

Comment:
is this really needed?

Reply:
The b & c tri color actually have different sizes so I'd assume so.

Reply:
no i meant the creation of an empty image each time ... there's already an image there

Reply:
The empty image probably isn't and would likely be fine if it was left to the Clear() function since colors should be cleared on startup. I left it since my changes are based off of someone else's fork and this was the point where things seemed to work reliably. Still ironing out the fine details and cleaning things up as I go.

Reply:
I am using it but i have to say the refresh rate is really painful. Empty image seems ok to avoid clear screen time.

Reply:
This is specific for epd2in13bc, you need to pass two arguments to displayupdate. One blackImage, and seconf with yellowredImage.
I guess, that empty image required to push display update is simple workaround to be backward compatible with other waveshare displays:

    def display(self, imageblack, imagered):
        self.send_command(0x10)
        for i in range(0, int(self.width * self.height / 8)):
            self.send_data(imageblack[i])
        self.send_command(0x92)
        
        self.send_command(0x13)
        for i in range(0, int(self.width * self.height / 8)):
            self.send_data(imagered[i])
        self.send_command(0x92)
        
        self.send_command(0x12) # REFRESH
        self.ReadBusy()

Reply:
So I put together a custom displayBlack function for the driver that basically omits any changes to the red/yellow color channels. Colors aren't not being used and are cleared out when pwnagotchi first boots anyways. Seems like it runs a little bit faster, so I'll be pushing up the changes soon.